src/test/java/plugins/GitLabPluginTest.java

```diff
@@ -49,307 +64,559 @@ public class GitLabPluginTest extends AbstractJUnitTest {
-        final WorkflowMultiBranchJob first_project =
-                organizationFolder.getJobs().get(WorkflowMultiBranchJob.class, groupName + "%2F" + repoName);
-        checksBuildsWithinProject(first_project);
+        organizationFolder.open();
+        waitFor().withTimeout(PAGE_LOAD_TIMEOUT).until(() -> driver.getPageSource()
+                .contains("Scan GitLab Group Now"));
 
-        // test the builds for the second project
-        final WorkflowMultiBranchJob second_project =
-                organizationFolder.getJobs().get(WorkflowMultiBranchJob.class, groupName + "%2F" + anotherRepoName);
-        checksBuildsWithinProject(second_project);
-    }
+        var projects = List.of(
+                organizationFolder.getJobs().get(WorkflowMultiBranchJob.class, GROUP_NAME + "%2F" + REPO_NAME),
+                organizationFolder.getJobs().get(WorkflowMultiBranchJob.class, GROUP_NAME + "%2F" + ANOTHER_REPO_NAME));
 
-    public void createRepo() throws RuntimeException {
-        // This sends a request to make a new repo in the gitlab server with the name "testrepo"
-        HttpResponse<String> response = container.createRepo(repoName, getPrivateTokenAdmin());
-        assertEquals(201, response.statusCode()); // 201 means the repo was created successfully
+        for (var project : projects) {
+            assertExistAndResult(project.getJob(MAIN_BRANCH), true);
+            assertExistAndResult(project.getJob(FIRST_BRANCH), true);
+            assertExistAndResult(project.getJob("MR-1"), true);
+            assertExistAndResult(project.getJob(FAILED_JOB), false);
+        }
     }
 
-    private void createBranch(GitLabApi gitlabapi, Project project, String branchName, String sourceBranch)
-            throws GitLabApiException {
-        gitlabapi.getRepositoryApi().createBranch(project.getId(), branchName, sourceBranch);
-    }
+    // ==================== Helper Methods ====================
 
-    private void addFile(GitLabApi gitlabapi, Project project, String branchName, String JenkinsfileContent)
-            throws GitLabApiException {
-        RepositoryFile file = new RepositoryFile();
-        file.setFileName("Jenkinsfile");
-        file.setFilePath("Jenkinsfile");
-        file.setContent(JenkinsfileContent);
+    /**
+     * GitLab uses "Scan GitLab Project Now" instead of generic "Scan Repository Now"
+     */
+    private void reIndex(WorkflowMultiBranchJob job) {
+        job.open();
+        // click the parent <a> tag
+        final List<WebElement> scanGitLabNow = driver.findElements(
+                by.xpath("//a[span[@class='task-link-text' and text()='Scan GitLab Project Now']]"));
 
-        // create Jenkinsfile on the main branch
-        gitlabapi.getRepositoryFileApi().createFile(project.getId(), file, branchName, "Add Jenkinsfile");
-    }
+        if (!scanGitLabNow.isEmpty()) {
+            scanGitLabNow.get(0).click();
+            return;
+        }
```

Comment:
what happens if the list is not empty (or the list has multiple matches?)

if you are only calling this with something from GitLab than use `findElement` and fail if the element is not found, otherwise if the element lookup fails to find a match the scan will not be triggered which could lead to either failing tests without an obvious reason, or worse flaky tests that are only working due to a periodic scan kicking in.